feature/35_SHOW_LISTING

Symphonie234 opened the pull request:
- Added functionality for liking and setting favorite of the listing
- Added a blade file for showing the content of the Listing

Symphonie234 committed:
added a functionality to reveal the content of the listing

Symphonie234 committed:
added functionality to show the contents of the listing and liking as well

Symphonie234 committed:
Merge branch 'develop' into feature/35_SHOW_LISTING

Symphonie234 committed:
added functionality for setting favorite

ChronoDevs commented on app/Models/Listing.php:
```diff
@@ -168,12 +168,22 @@ public function unlike(User $user)
-        } elseif ($likeCount <= 2) {
-            $likedBy = $this->likes->pluck('user.name')->implode(', ');
-            return "$likedBy liked this post";
+        } elseif ($likeCount === 1 && $this->likes->contains('user_id', $currentUser->id)) {
```
make an enum for $likeCount values

ChronoDevs replied:
also message use lang

ChronoDevs replied:
sorry the $likeCount condition should be check from enum
if ($likeCount === LikeCount::zero) { 

use your enum from 
class LikeCount extends Enum

ChronoDevs replied:
after fix. Please merge. Thank you

ChronoDevs commented on app/Http/Controllers/FavoriteController.php:
```diff
@@ -79,8 +79,9 @@ public function addToFavorites(Listing $listing)
         }
 
-        return redirect()->back();
+        return back()->with('error', 'Listing is already in favorites!');
```
use custom messages from lang

Symphonie234 committed:
attended to changes to be done in the PR

Symphonie234 commented:
### Changes that have been addressed in the PR No. 1:

- Created a new **Enums** file named **`LikeCount`**.
- Added a custom language for the **`FavoriteController`**.

### Additional Changes:

- Adjusted the **`FavoriteController`** and **`ListingController`**.
- Modified the **`Listing`** Model.
- Utilized **Spatie/Enum**.
- Included new aliases and providers in the **`config/auth.php`** file.
- Added a new JavaScript file named **`listings_show.js`** and incorporated new styles in the **`listings_show.css`**.
- Updated blade files to accommodate the changes.

Symphonie234 committed:
attended to the changes needed to be made in the PR #2

Symphonie234 commented:
### Changes that have been addressed in the PR No. 2:

- Created another **lang/en** file named **`listing.php`** to include specific messages for likes.
- Adjusted the code in the **`Listing`** Model accordingly.

Symphonie234 committed:
attended to the changed need to be changed in PR #3

Symphonie234 commented:
### Changes that have been addressed in the PR No. 3:

- Adjusted the code of the **`getLikesAttribute`** function in the **`Listing`** Model.
